### PR TITLE
Delete everything in repo-dir if we encounter an error

### PR DESCRIPTION
**Pull request by perryqh**
We are seeing .git/index.lock errors. We're unsure about the cause.

This change doesn't solve the root cause but deletes the repo-dir and starts over when it occurs.

If this doesn't work, we should seriously consider creating a brand new tmp repo-dir every time.

### CODE REVIEW

File: lib/github_bitbucket_deployer/git.rb [hunk 1]
```diff
@@ -5,23 +5,30 @@
...
-      push(remote, branch)
+      begin
+        logger.info('push_app_to_bitbucket')
+        add_remote(remote)
+        with_ssh { yield(repo) } if block_given?
+        push(remote, branch)
+      rescue => e
+        if force_pristine_repo_dir && !@already_forced_pristine_repo_dir
+          make_repo_dir_pristine
+          retry
```
Comment (perryqh) [inline lib/github_bitbucket_deployer/git.rb:27]: If we encounter an error, delete the repo-dir and `retry` one time

File: lib/github_bitbucket_deployer/git.rb [hunk 2]
```diff
@@ -34,7 +41,7 @@ def folder
 
     def clone
       logger.info("git clone: cloning #{bitbucket_repo_url} to #{folder}")
-      run { ::Git.clone(bitbucket_repo_url, folder, log: logger) }
+      run { ::Git.clone(bitbucket_repo_url, folder, log: logger, depth: 1) }
```
Comment (perryqh) [inline lib/github_bitbucket_deployer/git.rb:44]: we do not need history

File: lib/github_bitbucket_deployer/git.rb (hunk 1 above)
Comment (dpetersen) [reply lib/github_bitbucket_deployer/git.rb:27]: Literally never heard of this keyword.

**Review by dpetersen — APPROVED** [review]

File: lib/github_bitbucket_deployer/git.rb (hunk 2 above)
Comment (jcstringer) [reply lib/github_bitbucket_deployer/git.rb:44]: ➕ 1

File: lib/github_bitbucket_deployer/git.rb (hunk 1 above)
Comment (jcstringer) [reply lib/github_bitbucket_deployer/git.rb:27]: https://github.com/g5search/github_bitbucket_deployer/blob/master/github_bitbucket_deployer.gemspec#L20

**Review by jcstringer — APPROVED** [review]
🚢

**Review by markbluhm — APPROVED** [review]

**Pull request merged**